From ad70adf87a5cc7ea459835f432b013c5c13ec0e5 Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Mon, 9 Mar 2015 10:08:13 +0800
Subject: [PATCH] MLK-10367-02 tty: serial: imx: set uart module clock to
 80Mhz

When UART module clock is great than 80Mhz, there may have risk after
confirming with IC owner. So set the maximum module clock to 80Mhz.
(The patch just fix the first part that push un-correct patch version
by mistake)

Signed-off-by: Fugang Duan <B38611@freescale.com>
---
 drivers/tty/serial/imx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index b4db840..c2abd9e 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -2106,7 +2106,8 @@ static int serial_imx_probe(struct platform_device *pdev)
 		return ret;
 	}
 
-	if (sport->clk_per > IMX_MODULE_MAX_CLK_RATE)
+	sport->port.uartclk = clk_get_rate(sport->clk_per);
+	if (sport->port.uartclk > IMX_MODULE_MAX_CLK_RATE)
 		clk_set_rate(sport->clk_per, IMX_MODULE_MAX_CLK_RATE);
 	sport->port.uartclk = clk_get_rate(sport->clk_per);
 
-- 
1.8.0

